src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationRepository.cs

```diff
@@ -36,15 +37,27 @@ public async Task<List<MyPublicationViewModel>> GetAllPublicationsForTopicAsync(
 
         public async Task<List<MyPublicationViewModel>> GetPublicationsForTopicRelatedToUserAsync(Guid topicId, Guid userId)
         {
-            var userReleasesForTopic = await _context
-                .UserReleaseRoles
-                .Include(r => r.Release)
-                .ThenInclude(release => release.Publication)
-                .Where(r => r.UserId == userId && r.Release.Publication.TopicId == topicId && r.Role != ReleaseRole.PrereleaseViewer)
-                .Select(r => r.Release)
-                .Distinct()
+            var userReleasesForTopicFromReleases = await _context.UserReleaseRoles
+                .Include(userReleaseRole => userReleaseRole.Release.Publication)
+                .Where(userReleaseRole => userReleaseRole.UserId == userId &&
+                                          userReleaseRole.Release.Publication.TopicId == topicId &&
+                                          userReleaseRole.Role != ReleaseRole.PrereleaseViewer)
+                .Select(userReleaseRole => userReleaseRole.Release)
+                .ToListAsync();
+
+            var userReleasesForTopicFromPublications = await _context.UserPublicationRoles
+                .Where(userPublicationRole => userPublicationRole.UserId == userId &&
+                                              userPublicationRole.Publication.TopicId == topicId &&
+                                              userPublicationRole.Role == Owner)
+                .SelectMany(userPublicationRole => userPublicationRole.Publication.Releases)
+                .Include(release => release.Publication)
```

Comment:
This query wouldn't need it for the `.Where` because `release.Publication` isn't mentioned. It's also not needed because of the `.Where` of the query above for `userReleasesForTopicFromReleases` as far as I can tell, or at least removing it doesn't make it fail there.

It's needed further down once we've got the list of distinct releases and iterate over the list of distinct publications from those releases, where we `.Select(release => release.Publication)`. Without this `.include` the Publication can be null.

The plan is:
* Get all the distinct Releases they have access to either by Release role in `userReleasesForTopicFromReleases` and  by  Publication role in `userReleasesForTopicFromPublications`
* Get all the distinct Publications for those Releases - This needs the Publication to be included on the Releases.
* Build a collection of those Publications with their list of accessible Releases.

I can't move the `.Include` in `userReleasesForTopicFromReleases ` after the select as there's an EF Core rule that says includes are ignored if the query is returning a different entity than the one it began with. I think the update of the docs for EF Core 5 have removed this note that's applicable to our version so I can't link to it. I'm not sure why I'm able to include after the `.SelectMany` in the second case because I thought the same rule would apply, but I can, and it works. 🙂